doc/source/tutorial/interpolate/1D.rst

```diff
@@ -339,6 +341,55 @@ same data as in the previous example:
 ..             `next`.
 
 
+.. _tutorial-interpolate_interp1d_replacements:
+
+Recommended replacements for `interp1d` modes
+---------------------------------------------
+
+As mentioned, `interp1d` class is *legacy*: we have no plans to remove it; we
+are going to keep supporting its existing usages; however
+we believe there are better alternatives which we recommend using in new code.
+
+Here we list specific recommendations, depending on the interpolation ``kind``.
+
+Linear interpolation, ``kind="linear"``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default recommendation is to use `numpy.interp` function. Alternatively,
+you can use linear splines, ``make_interp_spline(x, y, k=1)``, see
+:ref:`this section for a discussion<tutorial-interpolate_linear_spline>`.
+
+
+Spline interpolators, ``kind="quadatic"`` or ``"cubic"``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Under the hood, ``interp1d`` delegates to ``make_interp_spline``, so we recommend
+using the latter directly.
+
+
+Piecewise constant modes, ``kind="nearest", "previous", "next"``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, we note that ``interp1d(x, y, kind='previous')`` is equivalent to
+``make_interp_spline(x, y, k=0)``.
+
+More generally however, all these piecewise constant interpolation modes are
+based on `numpy.searchsorted`. For example, the ``"nearest"`` mode is nothing but
+
+   >>> x = np.arange(8)
+   >>> y = x**2
+   >>> x_new = np.linspace(0, 7, 101)    # input points
+   >>> x_bds = x[:-1] / 2.0 + x[1:] / 2.0   # halfway points
+   >>> idx = np.searchsorted(x_bds, x_new, side='left')
+   >>> idx = np.clip(idx, 0, len(x) - 1)  # clip the indices so that they are within the range of x indices.
+   >>> import matplotlib.pyplot as plt
+   >>> plt.plot(x, y, 'o')
+   >>> plt.plot(x_new, y[idx], '--')
+   >>> plt.show()
+
+Other variants are similar, see the :ref:```interp1d`` source code <https://github.com/scipy/scipy/blob/v1.14.1/scipy/interpolate/_interpolate.py#L486>`
```

Comment:
This doesn't see to be rendering correctly 
![1000008105](https://github.com/user-attachments/assets/9bb30037-400b-4eca-87dc-40260ad297fd)